### ISSUE TITLE
What exactly is a Signal?

### ISSUE THREAD

**Issue by Urkman** [teslamotors/fleet-telemetry#266]
What is a signal and how they are counted?

And why do my Tesla send signal like this?
<img width="409" alt="Bildschirmfoto 2024-12-10 um 22 14 39" src="https://github.com/user-attachments/assets/89030f65-b59c-49ca-8620-08ca9e17fc78" />

Or all these changes while parked?
![Uploading Bildschirmfoto 2024-12-12 um 06.30.45.png…]()

All these problems will lead to why higher payments then necessary :(

**Comment by Urkman** [comment]
Here is the missing image...
<img width="388" alt="Bildschirmfoto 2024-12-12 um 06 30 45" src="https://github.com/user-attachments/assets/0c8d8804-8a0e-4aef-ad9d-9bad230b931a" />

**Comment by Bre77** [comment]
You have configured the vehicle to send you that data with `interval_seconds: 10`. If  you dont want to pay to get the data that often, then increase the interval. Unfortuantely floating point values like these change a lot.

I do agree that maybe sending gpsHeading and location when the vehicle is in park could be a little redundant, but hey, maybe its on the back of a tow truck?

**Comment by Urkman** [comment]
`Unfortuantely floating point values like these change a lot.... Unfortuantely floating point values like these change a lot.`
Sorry, they should not change back and forth and it should not send the same value again and again...

For other values like odometer this is not happening. For odometer I get exactly one signal when it is parked...

**Comment by jbanyer** [comment]
I would agree that it would make sense to have some sensible throttling on floating point value changes. Surely nobody cares about a 0.0001% change in battery SoC or energy remaining?

**Comment by Urkman** [comment]
And one more from my last drive.
These are 3 signals in one second, why are they not grouped?

<img width="399" alt="Bildschirmfoto 2024-12-12 um 08 26 51" src="https://github.com/user-attachments/assets/ddb60adb-e4f5-4981-9353-6c44b4b69764" />

**Comment by bassmaster187** [comment]
> And one more from my last drive. These are 3 signals in one second, why are they not grouped?

Even if they are grouped, they count as 3 signals.

**Comment by bassmaster187** [comment]
I count every single signal and that matches to the billing.

**Comment by bassmaster187** [comment]
Sorry, I am wrong. I was on par on 07th Dec., but it seems like today I'm far away from Tesla. Tesla should really make it clear. 

![image](https://github.com/user-attachments/assets/e88111fa-c837-4449-af56-438efeb189e0)

![image](https://github.com/user-attachments/assets/074b6fd3-f341-4276-bfd1-8c5567482f6c)

**Comment by dirkbonhomme** [comment]
Where can you track the number of signals and Tesla billing?

**Comment by Bre77** [comment]
> Where can you track the number of signals and Tesla billing?

The Fleet Telemetry server has a Statd and Prometheus metric output which you can use to count signals received. Your Tesla dashboard in billable countries shows your current billed volumes.

**Comment by bassmaster187** [comment]
> > Where can you track the number of signals and Tesla billing?
> 
> The Fleet Telemetry server has a Statd and Prometheus metric output which you can use to count signals received. Your Tesla dashboard in billable countries shows your current billed volumes.

I think I should update to the newest version. My istallation is from Feb. 2024 - never change an running system :-) 

Just override all files and restart the docker?

![image](https://github.com/user-attachments/assets/aad3791c-36d6-4082-bde6-4029c859ae9e)

**Comment by Urkman** [comment]
> Tesla should really make it clear.

Yes, that is a huge problem. 
No support and no documentation :( But the billing is fine...

**Comment by patrickdemers6** [comment]
A signal is one field's value delivered to your server. 3 fields in one message counts as 3 signals.

Please submit a support ticket if you still don't think we're tracking usage properly @bassmaster187.

> I would agree that it would make sense to have some sensible throttling on floating point value changes. Surely nobody cares about a 0.0001% change in battery SoC or energy remaining?

minimum_delta is now available and provides exactly this.
